### PR TITLE
Enable IDE0049: PreferBuiltInOrFrameworkType

### PR DESCRIPTION
**Pull request by xtqqczze**
cc: @iSazonov

https://docs.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0049

Skipped:
`src\Microsoft.PowerShell.Commands.Diagnostics\CounterSample.cs`
`src\Microsoft.PowerShell.Commands.Diagnostics\GetCounterCommand.cs`
`src\Microsoft.PowerShell.Commands.Diagnostics\PdhHelper.cs`

### CODE REVIEW

**Comment by iSazonov** [comment]
I doubt it is necessary. These CIM, WSMan, Counter codes are specific and frozen. I think we should only set the severity to suggestion.

**Comment by adityapatwardhan** [comment]
@xtqqczze Please resolve merge conflicts

**Comment by anmenaga** [comment]
Maintainer review summary: ok to take this considering that this does not touch files that are frequently changed therefore the chance of introducing merge conflicts is low. Additionally, files that are not part of the build should be removed from this PR (@adityapatwardhan to give list of such files).

**Comment by adityapatwardhan** [comment]
The files that are not compiled and hence should be removed from PR. Files to be removed:

`PdhHelper.cs`
`CounterSample.cs`
`GetCounterCommand.cs`

**Comment by xtqqczze** [comment]
In `dotnet/roslyn`, the IDE0049 rule is specified as `EnforceOnBuild.Never` therefore we cannot enforce the rule during build, see:
https://github.com/dotnet/roslyn/issues/50173
https://github.com/dotnet/roslyn/blob/9c10b941e96760e0c986c8f2afb0d5b4c34e67e3/src/Analyzers/Core/Analyzers/EnforceOnBuildValues.cs#L105

There is still value in enabling the rule, as it will be used for live analysis in the IDE.

**Comment by xtqqczze** [comment]
@adityapatwardhan The suggested changes have been made, please could you continue your review.

File: src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs [hunk 1]
```diff
@@ -587,14 +587,13 @@ private List<string> CombineMachinesAndCounterPaths()
...
-                        string slashBeforePath = path.Length > 0 && path[0] == '\\' ? string.Empty : "\\";
                         if (machine.StartsWith("\\\\", StringComparison.OrdinalIgnoreCase))
                         {
-                            retColl.Add(machine + slashBeforePath + path);
+                            retColl.Add(machine + "\\" + path);
                         }
                         else
                         {
-                            retColl.Add("\\\\" + machine + slashBeforePath + path);
+                            retColl.Add("\\\\" + machine + "\\" + path);
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Diagnostics/GetCounterCommand.cs:596]: Please revert the unrelated change.

**Comment by xtqqczze** [comment]
rebased

**Review by iSazonov — APPROVED** [review]

**Review by rkeithhill — APPROVED** [review]
LGTM. I like the consistency even though it is a lot of code change.

**Review by adityapatwardhan — APPROVED** [review]

**Comment by adityapatwardhan** [comment]
@xtqqczze Thank you for your contribution!

**Comment by xtqqczze** [comment]
Contributes to https://github.com/PowerShell/PowerShell/issues/25922.
